v2/internal/reconcilers/arm/arm_client_cache.go

```diff
@@ -115,7 +116,7 @@ func (c *ARMClientCache) GetClient(ctx context.Context, obj genruntime.ARMMetaOb
 }
 
 func (c *ARMClientCache) getPerResourceCredential(ctx context.Context, obj genruntime.ARMMetaObject) (*armClient, error) {
-	return c.getCredentialFromAnnotation(ctx, obj, PerResourceSecretAnnotation)
+	return c.getCredentialFromAnnotation(ctx, obj, reconcilers.PerResourceSecretAnnotation)
```

Comment:
Could the same be for namespacedCredential, if we're not watching the namespaced secret?

Reply:
There were 2 separate issues actually:
1. When an update was made to an object that changed a property of that object which we were not watching (for example an annotation we are ignoring) it was possible if the timing was right that the ongoing reconcile of that object was interrupted but a new one wasn't queued. This would cause us to get "stuck" at least for a while.
2. We were ignoring changes to this annotation.

1 was the "bigger" bug, as it can happen with any annotation, including user annotations that we'll never care about.
2 was just an easy additional fix, because we don't have to be watching the corresponding secret to detect changes, we're already watching our own object we just need to stop ignoring changes to this annotation.

Watching the namespace scoped secret is harder because it requires setting up a watch, etc. We can do it but it's not going to cause a bug if we don't.